src/code_translation/ir/nodes.py

```diff
@@ -90,6 +94,15 @@ def __init__(
 
         super().__init__(*args, **kwargs)
 
+    @property
+    def con(self) -> cst.CSTNode:
+        left_con = self.left.con
+        right_con = self.right.con
+        if left_con.value != right_con.value:
+            raise Exception("Join partners from different sources")
+        else:
+            return left_con
```

Comment:
I wanted to check this anyway for some time, because when I looked at some of the example pipelines from Ilin, I saw that often in one notebook multiple connections are used.
This obviously does not quiet work if you have something like:
`con1 = DBConnection(...)` and `con2 = DBConnection(...)`, so the same connection with 2 different names. Not sure how relevant this is for us though

Reply:
We can assume we have the same connection, but I won't delete your code, since it's already written ^^